compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -154,22 +154,24 @@ object DesugarEnums {
   private def enumLookupMethods(constraints: EnumConstraints)(using Context): List[Tree] =
     def scaffolding: List[Tree] = if constraints.cached then enumScaffolding(constraints.enumCases.map(_._2)) else Nil
     def valueCtor: List[Tree] = if constraints.requiresCreator then enumValueCreator :: Nil else Nil
-    def byOrdinal: List[Tree] =
-      if isJavaEnum || !constraints.cached then Nil
+    def fromOrdinal: Tree =
+      def throwArg(ordinal: Tree) =
+        Throw(New(TypeTree(defn.NoSuchElementExceptionType), List(Select(ordinal, nme.toString_) :: Nil)))
+      if !constraints.cached then
+        fromOrdinalMeth(throwArg)
       else
-        val defaultCase =
-          val ord = Ident(nme.ordinal)
-          val err = Throw(New(TypeTree(defn.IndexOutOfBoundsException.typeRef), List(Select(ord, nme.toString_) :: Nil)))
-          CaseDef(ord, EmptyTree, err)
-        val valueCases = constraints.enumCases.map((i, enumValue) =>
-          CaseDef(Literal(Constant(i)), EmptyTree, enumValue)
-        ) ::: defaultCase :: Nil
-        val fromOrdinalDef = DefDef(nme.fromOrdinalDollar, Nil, List(param(nme.ordinalDollar_, defn.IntType) :: Nil),
-          rawRef(enumClass.typeRef), Match(Ident(nme.ordinalDollar_), valueCases))
-            .withFlags(Synthetic | Private)
-        fromOrdinalDef :: Nil
-
-    scaffolding ::: valueCtor ::: byOrdinal
+        def default(ordinal: Tree) =
+          CaseDef(Ident(nme.x_0), EmptyTree, throwArg(ordinal))
```

Comment:
If the name isn't used then wildcard is more appropriate:
```suggestion
          CaseDef(Ident(nme.WILDCARD), EmptyTree, throwArg(ordinal))
```

Reply:
thanks for the tip